File: mobile/lib/models/Recipe.dart
import 'ingredient.dart';
class Recipe{
  String? name;
  String? instructions;
  List<Ingredient>? ingredients;
  List<String>? tags;
  Recipe();
  @override
  String toString() {
    return 'Recipe $name, instructions $instructions, ingredients ${ingredients.toString()}, tags $tags';
  }
  Recipe.fromJson(Map<String, dynamic>? json){
    if(json==null)  return;
    name = json['name'];
    instructions = json['instructions'];
    ingredients = Ingredient.listFromJson(json['ingredients']);
    //TODO::correct tags, probably won't work. do I have to initialize tags?
    for(String? tag in json['tags'])
    {
      if(tag!=null){
        tags?.add(tag);
      }
    }
  }
}

File: Backend/Cookly/Cookly/src/main/java/com/example/Cookly/Models/Recipe.java
public class Recipe {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;
    @Column(nullable = false) private String name;
    @Column(nullable = false) private String instructions;
    @Column(nullable = false, name = "ingredient_id")
    private List<List<Ingredient>, String> ingredients;
    @Column(nullable = false, name = "tag")
    private List<String> tags;
}